decisions/2019/04/11.md

```diff
@@ -0,0 +1,46 @@
+# Front-end Decision Board, 2019-04-11
+
+## #31 Validation of config objects
+
+**Conclusion**: This is inherently a problem that depends on the
+context.
+
+The general rule for parameters is the relative standard:
+
+- Less than 3 params can be positional
+- 3 and above warrants a configuation object.
+
+Validation should be considered compile-time and runtime for the
+required properties. Should be flexible and accept parameters which are
+unknown to the function.
+
+Mixing configuration objects and parameters? Ultimately a judgement
+call. Sometimes it makes sense to have e.g. `fetchUrl('google.com',
+{})`. Be mindful during code reviews.
+
+## #23 Which future syntax do we want to use in our js?
+
+**Conclusion**: Stage-4, with few exceptions (based on the
+react-scripts), for example class-based-props.
+
+New exceptions should be handled with care through discussion with the
+frontend group.
+
+Be aware of logical ordering of class properties, until we are able to
+delint this.
+
+Avoid decorators where possible.
+
+## #15 One file for one function
+
+**Conclusion**: "It depends". Ultimately it is a judgement call for the
+developers and reviewers. It can have a big impact on understanding a
+code base, so do keep it in mind to err on the side of readability.
+Files are cheap as Git only tracks content.
+
+Generally smaller files are better. Use common sense as to what makes
+the code easier to read and understand.
+
+Utility files is one example where if every util function is kept in the
+same file it becomes a mess and hard to read and keep track of.
+Divide and conquer.
```

Comment:
Optionally add a sentence like this:
> Never export more than one independent component from a single file

(I mean components that are actually meant be used independently, it's OK if some internal/helper component is exported from a file for unit-testing)